### PR TITLE
Fixes AI interactions

### PR DESCRIPTION
**Pull request by psykzz**
Fixes AI not being able to use pretty much anything.
Previously it was doing an old "range" check instead of a cam vis check.

Also adds some missing flags.

### CODE REVIEW

File: code/_onclick/ai.dm [hunk 1]
```diff
@@ -19,12 +19,14 @@
 		A.move_camera_by_click()
 
 
-/mob/living/silicon/ai/ClickOn(atom/A, location, params)
+/mob/living/silicon/ai/ClickOn(src, location, params)
```
Comment (psykzz) [inline code/_onclick/ai.dm:22]:
```suggestion
/mob/living/silicon/ai/ClickOn(atom/A, location, params)
```
ops

File: code/_onclick/ai.dm [hunk 2]
```diff
@@ -42,15 +44,23 @@
 	var/turf/pixel_turf = get_turf_pixel(A)
 	if(isnull(pixel_turf))
 		return
-
-	if(!can_see(A) && isturf(A))
+	if(!can_see(A))
+		if(isturf(A)) //On unmodified clients clicking the static overlay clicks the turf underneath
+			return //So there's no point messaging admins
+		message_admins("[ADMIN_LOOKUPFLW(src)] might be running a modified client! (failed can_see on AI click of [A] (Turf Loc: [ADMIN_VERBOSEJMP(pixel_turf)]))")
+		var/message = "[key_name(src)] might be running a modified client! (failed can_see on AI click of [A] (Turf Loc: [AREACOORD(pixel_turf)]))"
```
Comment (TiviPlus) [inline code/_onclick/ai.dm:51]: There is currently a bug on both tg and tgmc that spuriously allows ais to see the whole map so this isnt 100% accurate, should elaborate that that might also be the case

Comment (psykzz) [reply code/_onclick/ai.dm:51]: I dont think there is a need to make the string longer, most of the time when this happens admins either ask each other or ping me about it.

Comment (TiviPlus) [reply code/_onclick/ai.dm:51]: Okey dokey then

**Pull request merged**